### ISSUE TITLE
MultiGPU FFTN

### ISSUE THREAD

**Issue by Jackson-H-Code** [cupy/cupy#7732]
### Description

Howdy. I am curious about the current level of support for higher-dimensional multigpu ffts in CuPy. It appears to me that only 1D ffts are currently supported through cp.cuda.cufft.Plan1d when creating multigpu ffts for computations that are too large to fit on one gpu. I was wondering if I am just missing that there currently is support for PlanNd on multiple gpus, or if it is either being worked on actively?

Thanks so much!

**Comment by leofang** [comment]
No, I don't think anyone is working on multi-GPU Nd FFT support, feel free to give it a shot. My advices:
1. Ignore `PlanNd` and just focus on `XtPlanNd` (so that it can naturally work for large-array transforms, see https://github.com/cupy/cupy/issues/4406)
2. The existing support in `Plan1d` can be used as a reference, though I expect there are additional work with Nd FFT.

**Comment by dialecticDolt** [comment]
A main annoyance in this will be that `cufftXtMemcpy` only supports 1D data for D2D so transforms on 2D/3D data will need to be either driven entirely from the host and back after each computation (as some of the current CuPy multigpu-FFT support is currently implemented) or the wrapper will need its own reshuffling back to the original input order. 

I wrote a Cython wrapper for the 2D multigpu-cuFFT for my own purposes recently, so I might start poking around to see if I can extend the wrappers here.  No promises on the time or effort.. it might be a fair bit before I have the cycles to spare.
